docs/integration-testing.md

```diff
@@ -0,0 +1,86 @@
+Integration testing is conducted to evaluate the compliance of a system or component with specified functional requirements. 
+It occurs after unit testing and before system testing. Integration testing takes as its input modules that have been unit tested, groups them in larger aggregates, applies tests defined in an integration test plan to those aggregates, and delivers as its output the integrated system ready for system testing.
+
+To group these integration test aggregates, we make use of test suites. In software development, a test suite, less commonly known as a validation suite, is a collection of test cases that are intended to be used to test a software program to show that it has some specified set of behaviours. 
+A test suite often contains detailed instructions or goals for each collection of test cases and information on the system configuration to be used during testing. 
+A group of test cases may also contain prerequisite states or steps, and descriptions of the following tests.
+
+## Database Integration Testing (Postgres)
+
+To run our test suite we use the go library [testify](https://pkg.go.dev/github.com/stretchr/testify/suite), specifically the suite capability. 
+Compared to regular testing, this enables us to manage lifecycle of tests run inside a suite through a set of predefined methods (for example, run before or after all or individual tests are executed). 
+As a result, it enables us to manage dependant components of the reconciler, in this case our database.
+
+### How to write a Postgres Integration Test
+
+To write your own test based on Postgres, there is a helper available that you can use:
+`pkg/db/transaction_aware_database_container_test_suite.go`
+
+This suite is able to manage a container lifecycle of a postgres database with the help of the [testcontainers](https://golang.testcontainers.org/) framework, enabling us to couple independent database containers with our test suite execution. 
+It provides the ability to isolate tests running together with the database.
+
+To write your own tests, you won't have to set the test suite up yourself, but you can use established helpers depending on your use case.
+
+#### Shared Container Integration Tests with automatic Rollback
+
+This setup is nice if you just want to run some business logic of the reconciler against a postgres-database and within one transaction context. 
+The test suite takes care of using a shared container that is reused between suites with the same isolation needs and rolls back your changes made in the test instead of committing them automatically.
+
+The reuse can be achieved by "leasing" a container suite from a globally managed map.
+
+*IMPORTANT: You must always return the Lease of the container suite so that the container is cleaned up correctly with `ReturnLeasedSharedContainerTestSuite(*testing.T, settings ContainerSettings)`*
```

Comment:
```suggestion
> **IMPORTANT**: You must always return the lease of the container suite so that the container is cleaned up correctly with `ReturnLeasedSharedContainerTestSuite(*testing.T, settings ContainerSettings)`.
```